frontend/app/src/utils/sns.ts

```diff
@@ -11,3 +11,8 @@ export function isPrincipalValid(text: string): boolean {
     }
 }
 
+const SUB_ACCOUNT_REGEX = new RegExp("^[A-Fa-f0-9]{64}$");
```

Comment:
Is this saying it has to be exactly 64 chars?
If so that is wrong, you are allowed to trim the 0's from the left.
So subaccount '1' is valid, we simply need to pad it with 0's